src/main.go

```diff
@@ -158,8 +181,18 @@ func handleToken(c echo.Context) error {
 	cookie.HttpOnly = true
 	c.SetCookie(cookie)
 	email := RROnline.GetEmail(userTok)
-	if !RROnline.UserExists(email) {
-		db := RROnline.Connect()
+	exists, dbErr := RROnline.UserExists(email)
+	if dbErr != nil {
+		RROnline.LogDBError(dbErr)
+	}
+	if !exists {
+		db, err := RROnline.Connect()
```

Comment:
Failure to connect to DB should be a fatal event in the thread. Should result in 5XX code or some default page saying something isn't working